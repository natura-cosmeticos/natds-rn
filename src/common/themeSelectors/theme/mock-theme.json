{
  "avatarSizes": {
    "tiny": {
      "size": 24,
      "fontSize": ".625rem"
    },
    "small": {
      "size": 32,
      "fontSize": ".875rem"
    },
    "standard": {
      "size": 40,
      "fontSize": "1rem"
    },
    "large": {
      "size": 48,
      "fontSize": "1rem"
    },
    "huge": {
      "size": 80,
      "fontSize": "1.5rem"
    }
  },
  "buttonSizes": {
    "small": {
      "paddingTop": 8,
      "paddingRight": 8,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "minHeight": 32
    },
    "medium": {
      "paddingTop": 12,
      "paddingRight": 16,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "minHeight": 40
    },
    "large": {
      "paddingTop": 16,
      "paddingRight": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "minHeight": 48
    }
  },
  "fontSize": {
    "h1": {
      "fontSize": "6rem",
      "fontWeight": 300
    },
    "h2": {
      "fontSize": "3.75rem",
      "fontWeight": 300
    },
    "h3": {
      "fontSize": "3rem",
      "fontWeight": 400
    },
    "h4": {
      "fontSize": "2.125rem",
      "fontWeight": 400
    },
    "h5": {
      "fontSize": "1.5rem",
      "fontWeight": 400
    },
    "h6": {
      "fontSize": "1.25rem",
      "fontWeight": 600
    },
    "subtitle1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "subtitle2": {
      "fontSize": "0.875rem",
      "fontWeight": 500
    },
    "body1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "body2": {
      "fontSize": "0.875rem",
      "fontWeight": 400
    },
    "button": {
      "fontSize": "0.875rem",
      "fontWeight": 600
    },
    "caption": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    },
    "overline": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    }
  },
  "iconSizes": {
    "micro": 16,
    "tiny": 24,
    "small": 32,
    "standard": 48,
    "medium": 64,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "huge": 144,
    "hugex": 192,
    "hugexx": 256
  },
  "opacity": {
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
    "00": 0,
    "04": 0.04,
    "08": 0.08
  },
  "overlay": {
    "solidBlack": 0.56
  },
  "radius": {
    "none": 0,
    "circle": "50%",
    "small": 2,
    "medium": 4,
    "large": 8
  },
  "sizes": {
    "none": 0,
    "micro": 4,
    "tiny": 8,
    "small": 16,
    "standard": 24,
    "semi": 32,
    "semix": 40,
    "medium": 48,
    "mediumx": 56,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "veryhuge": 256
  },
  "spacing": {
    "spacing": 8,
    "spacingNone": 0,
    "spacingMicro": 4,
    "spacingTiny": 8,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingSemi": 32,
    "spacingLarge": 48,
    "spacingHuge": 64
  },
  "colorTokens": {
    "colorPrimary": "#F4AB34",
    "colorOnPrimary": "#333333",
    "colorPrimaryLight": "#FEFDE8",
    "colorOnPrimaryLight": "#333333",
    "colorPrimaryDark": "#EF8426",
    "colorOnPrimaryDark": "#333333",
    "colorSecondary": "#FF6B0B",
    "colorOnSecondary": "#000000",
    "colorSecondaryLight": "#FFF8E1",
    "colorOnSecondaryLight": "#000000",
    "colorSecondaryDark": "#FF5808",
    "colorOnSecondaryDark": "#000000",
    "colorBackground": "#FAFAFA",
    "colorOnBackground": "#333333",
    "colorSurface": "#FFFFFF",
    "colorOnSurface": "#333333",
    "colorHighlight": "#000000",
    "colorHighEmphasis": "#333333",
    "colorMediumEmphasis": "#777777",
    "colorLowEmphasis": "#BBBBBB",
    "colorLink": "#227BBD",
    "colorOnLink": "#FFFFFF",
    "colorSuccess": "#569A32",
    "colorOnSuccess": "#FFFFFF",
    "colorWarning": "#FCC433",
    "colorOnWarning": "#333333",
    "colorAlert": "#E74627",
    "colorOnAlert": "#FFFFFF"
  },
  "elevation": {
    "1": {
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 1, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 1
    },
    "2": {
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 2, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 2
    },
    "3": {
      "elevation": 3,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 3, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 4
    },
    "4": {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 4, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 5
    },
    "5": {
      "elevation": 5,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 6, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "6": {
      "elevation": 6,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 8, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "7": {
      "elevation": 7,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 9, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 12
    },
    "8": {
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 12, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 17
    },
    "9": {
      "elevation": 9,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 16, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 24
    },
    "10": {
      "elevation": 10,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 24, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 38
    }
  }
}
