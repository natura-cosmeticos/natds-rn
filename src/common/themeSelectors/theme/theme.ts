import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
