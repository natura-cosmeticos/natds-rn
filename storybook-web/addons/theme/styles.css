.theme {
  font-weight: 400;
  line-height: 1.5;
  max-width: 800px;
  font-family: Roboto, Arial, sans-serif;
  color: #555;
  margin: 30px;
}

.theme__item {
  max-width: 600px;
  border-bottom: rgba(0,0,0,.1) 1px solid;
  display: flex;
  align-items: center;
  padding: 0 0 10px;
  margin: 0 0 10px;
}

.theme__name {
  flex: 2 0 auto;
  font-size: 18px;
  text-transform: capitalize;
  padding: 0;
}

.theme__button {
  background: rgba(0,0,0,.3);
  border: 0;
  border-radius: 4px;
  font-size: 1rem;
  color: #fff;
  text-transform: capitalize;
  cursor: pointer;
  padding: 7px 14px;
  margin: 0 10px 0 0;
}

.theme__button:hover {
  background: rgba(0,0,0,.2);
}

.theme__button:active {
  background: rgba(0,0,0,.25);
}

.theme__button--selected,
.theme__button--selected:hover {
  background: rgba(0,0,0,.6);
  cursor: default;
}
