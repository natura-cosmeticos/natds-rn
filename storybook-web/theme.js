/* eslint-disable sort-keys */
import { create } from '@storybook/theming/create';
import themes from '@naturacosmeticos/natds-themes/react-native';

const buildTheme = (theme, mode) => create({
  base: mode,
  colorPrimary: theme.color.primary,
  colorSecondary: theme.color.secondary,

  // UI
  appBg: theme.color.surface,
  appContentBg: theme.color.background,

  // Text
  textColor: theme.color.highEmphasis,
  textInverseColor: theme.color.lowEmphasis,

  // Toolbar
  barBg: theme.color.surface,
  barTextColor: theme.color.highEmphasis,

  // Brand
  brandTitle: 'Natura Design System',
});

export const dark = buildTheme(themes.natura.dark, 'dark');
export const light = buildTheme(themes.natura.light, 'light');
