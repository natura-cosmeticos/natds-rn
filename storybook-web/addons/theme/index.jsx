import React, { useEffect, Fragment } from 'react';
import { useAddonState, Consumer } from '@storybook/api';
import { TabButton, WithTooltip, TooltipLinkList } from '@storybook/components';
import {
  PANEL_ID, CHANGE, PARAM_KEY, getStoryBookTheme,
} from './shared';
import { light, dark } from '../../theme';

import './styles.css';

const DEFAULT_THEME = {
  name: 'natura',
  type: 'light',
};

const buildThemeProps = mode => ({ ...DEFAULT_THEME, type: mode });

function parseTheme(themes) {
  return Object.entries(themes).reduce((accum, [name, variants]) => {
    Object.entries(variants).forEach(([type]) => {
      accum.push({ name, type });
    });

    return accum;
  }, []);
}

const createBackgroundSelectorItem = ({ name, type }, onChange, selectedItem) => ({
  active: selectedItem.name === name && selectedItem.type === type,
  id: `${name} - ${type}`,
  onClick: () => onChange({ name, type }),
  title: `${name} - ${type}`,
});

const getDisplayedItems = (list, onChange, selectedItem) => {
  if (!list.length) {
    return [];
  }

  return [
    ...list.map(params => createBackgroundSelectorItem(params, onChange, selectedItem)),
  ];
};

const mapper = ({ api, state }) => {
  const story = state.storiesHash[state.storyId];
  const { themes = {} } = (story && api.getParameters(story.id, PARAM_KEY)) || {};

  return { items: parseTheme(themes) };
};

export default function Theme({ channel, api }) {
  const apiTheme = getStoryBookTheme();
  const initialTheme = apiTheme ? buildThemeProps(apiTheme) : DEFAULT_THEME;
  const [currentTheme, changeTheme] = useAddonState(PANEL_ID, initialTheme);
  const { disabled } = api.getCurrentParameter(PARAM_KEY) || {};

  const handleChange = (params) => {
    changeTheme(params);
    api.setOptions({
      theme: params.type === 'light' ? light : dark,
    });
    const event = new Event('CHANGE_THEME', { detail: params.type });

    window.dispatchEvent(event);
    channel.emit(CHANGE, params);
  };

  const renderTooltip = items => ({ onHide }) => (
    <TooltipLinkList
      links={getDisplayedItems(items, (index) => {
        handleChange(index);
        onHide();
      }, currentTheme)}
    />
  );

  const eventListener = event => handleChange(buildThemeProps(event.matches ? 'dark' : 'light'));

  useEffect(() => {
    handleChange(initialTheme);

    const matchMediaEnvent = window.matchMedia('(prefers-color-scheme: dark)');

    matchMediaEnvent.addEventListener('change', eventListener);

    return () => {
      matchMediaEnvent.removeEventListener('change', eventListener);
    };
  }, []);

  if (disabled) return null;

  return (
    <Consumer filter={mapper}>
      {({ items }) => (items.length ? (
          <Fragment>
            <WithTooltip
              placement="top"
              trigger="click"
              closeOnClick
              tooltip={renderTooltip(items)}
            >
              <TabButton
                key="Theme"
                title="Change the Theme">
                {currentTheme.name} - {currentTheme.type}
              </TabButton>
            </WithTooltip>
          </Fragment>
      ) : null)}
    </Consumer>
  );
}
