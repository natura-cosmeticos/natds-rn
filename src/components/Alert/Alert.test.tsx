import React from 'react'
import theme from '../../common/themeSelectors/theme/mock-theme.json'
import { Alert } from '.'
import { renderWithTheme } from '../../../test/testHelpers'

jest.mock('../../common/themeSelectors', () => (
  {
    getColorByName: () => 'red',
    getColorOnPrimary: () => '#333333',
    getRadiusBySize: () => 8,
    getSize: () => 24,
    getSpacingSmall: () => 16,
    getSpacingTiny: () => 8
  }))

describe('Alert component', () => {
  it('Should render alert with default props', () => {
    const { toJSON } = renderWithTheme(<Alert testID="alert" title="title" message="Alert text message" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('Should render alert with outlined variant', () => {
    const { toJSON } = renderWithTheme(<Alert variant="outlined" title="title" message="Alert text message" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('Should render alert with custom type with outlined variant', () => {
    const { toJSON } = renderWithTheme(
      <Alert
        type="custom"
        title="title"
        message="Alert text message"
        variant="outlined"
        backgroundColorName="primary"
        borderColorName="primary"
        iconName="filled-action-check"
        iconColorName="link"
        messageColorName="link"
        titleColorName="link"
      />
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('Should change icon when type changes', () => {
    const { queryByTestId } = renderWithTheme(<Alert type="success" title="title" message="Alert text message" />)

    expect(queryByTestId('alert-icon')?.props?.children?.props).toHaveProperty('name', 'outlined-alert-check')
  })

  it('Should change alert background color when type changes', () => {
    const { queryByTestId } = renderWithTheme(<Alert type="warning" title="title" message="Alert text message" />)

    expect(queryByTestId('alert')?.props?.style[0]).toHaveProperty('backgroundColor', `${theme.color.warning}29`)
  })

  it('Should hide title component when not informed', () => {
    const { queryByTestId } = renderWithTheme(<Alert message="Alert text message" />)

    expect(queryByTestId('alert-content')?.props?.children[0]).toBeFalsy()
  })

  it('Should hide icon component when icon props is false', () => {
    const { queryByTestId } = renderWithTheme(<Alert icon={false} title="title" message="Alert text message" />)

    expect(queryByTestId('alert-icon')?.props?.children).toBeFalsy()
  })
})
