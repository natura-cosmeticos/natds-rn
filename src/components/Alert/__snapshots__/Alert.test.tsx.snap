// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component Should render alert with custom type with outlined variant 1`] = `
<View
  backgroundColorName="primary"
  borderColorName="primary"
  iconColorName="link"
  iconName="filled-action-check"
  message="Alert text message"
  messageColorName="link"
  style={
    Array [
      Object {
        "backgroundColor": "#f4ab3429",
        "borderColor": "#f4ab34",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
  testID="alert"
  title="title"
  titleColorName="link"
  type="custom"
  variant="outlined"
>
  <View
    style={
      Array [
        Object {
          "marginRight": 8,
          "marginTop": 2,
        },
      ]
    }
    testID="alert-icon"
  >
    <Text
      accessibilityRole="imagebutton"
      color="link"
      size="standard"
      style={
        Array [
          Object {
            "color": "red",
            "fontFamily": "natds-icons",
            "fontSize": 24,
          },
        ]
      }
      testID="natds-icon"
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexShrink": 1,
        },
      ]
    }
    testID="alert-content"
  >
    <Text
      backgroundColorName="primary"
      borderColorName="primary"
      iconColorName="link"
      iconName="filled-action-check"
      message="Alert text message"
      messageColorName="link"
      style={
        Array [
          Object {
            "color": "#227bbd",
            "flexWrap": "wrap",
            "fontFamily": "Roboto",
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0.24,
            "lineHeight": 30,
            "marginBottom": 8,
          },
        ]
      }
      title="title"
      titleColorName="link"
      type="custom"
      variant="outlined"
    >
      title
    </Text>
    <Text
      backgroundColorName="primary"
      borderColorName="primary"
      iconColorName="link"
      iconName="filled-action-check"
      message="Alert text message"
      messageColorName="link"
      style={
        Array [
          Object {
            "color": "#227bbd",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.51,
            "lineHeight": 24,
          },
        ]
      }
      title="title"
      titleColorName="link"
      type="custom"
      variant="outlined"
    >
      Alert text message
    </Text>
  </View>
</View>
`;

exports[`Alert component Should render alert with default props 1`] = `
<View
  message="Alert text message"
  style={
    Array [
      Object {
        "backgroundColor": "#227bbd29",
        "borderRadius": 4,
        "borderWidth": 0,
        "flexDirection": "row",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
  testID="alert"
  title="title"
  type="info"
  variant="standard"
>
  <View
    style={
      Array [
        Object {
          "marginRight": 8,
          "marginTop": 2,
        },
      ]
    }
    testID="alert-icon"
  >
    <Text
      accessibilityRole="imagebutton"
      color="neutral900"
      size="standard"
      style={
        Array [
          Object {
            "color": "red",
            "fontFamily": "natds-icons",
            "fontSize": 24,
          },
        ]
      }
      testID="natds-icon"
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexShrink": 1,
        },
      ]
    }
    testID="alert-content"
  >
    <Text
      message="Alert text message"
      style={
        Array [
          Object {
            "color": "#212529",
            "flexWrap": "wrap",
            "fontFamily": "Roboto",
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0.24,
            "lineHeight": 30,
            "marginBottom": 8,
          },
        ]
      }
      testID="alert"
      title="title"
    >
      title
    </Text>
    <Text
      message="Alert text message"
      style={
        Array [
          Object {
            "color": "#212529",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.51,
            "lineHeight": 24,
          },
        ]
      }
      testID="alert"
      title="title"
    >
      Alert text message
    </Text>
  </View>
</View>
`;

exports[`Alert component Should render alert with outlined variant 1`] = `
<View
  message="Alert text message"
  style={
    Array [
      Object {
        "backgroundColor": "#227bbd29",
        "borderColor": "#227bbd",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
  testID="alert"
  title="title"
  type="info"
  variant="outlined"
>
  <View
    style={
      Array [
        Object {
          "marginRight": 8,
          "marginTop": 2,
        },
      ]
    }
    testID="alert-icon"
  >
    <Text
      accessibilityRole="imagebutton"
      color="neutral900"
      size="standard"
      style={
        Array [
          Object {
            "color": "red",
            "fontFamily": "natds-icons",
            "fontSize": 24,
          },
        ]
      }
      testID="natds-icon"
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexShrink": 1,
        },
      ]
    }
    testID="alert-content"
  >
    <Text
      message="Alert text message"
      style={
        Array [
          Object {
            "color": "#212529",
            "flexWrap": "wrap",
            "fontFamily": "Roboto",
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0.24,
            "lineHeight": 30,
            "marginBottom": 8,
          },
        ]
      }
      title="title"
      variant="outlined"
    >
      title
    </Text>
    <Text
      message="Alert text message"
      style={
        Array [
          Object {
            "color": "#212529",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.51,
            "lineHeight": 24,
          },
        ]
      }
      title="title"
      variant="outlined"
    >
      Alert text message
    </Text>
  </View>
</View>
`;
